### ISSUE TITLE
`rustdoc --passes=list` doesn't show a deprecation warning

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#91713]
I tried this command: `rustdoc --passes=list`

I expected to see this happen: rustdoc shows a warning, the way it does for any other value you pass to `--passes`:

```
$ rustdoc +nightly --passes check-invalid-html-tags
warning: the `passes` flag is deprecated
  |
  = note: see issue #44136 <https://github.com/rust-lang/rust/issues/44136> for more information
```

Instead, this happened: Rustdoc gives no warning and lists all passes.

### Meta

`rustdoc --version`
```
rustdoc 1.58.0-nightly (e90c5fbbc 2021-11-12)
```

**Comment by jyn514** [comment]
Mentoring instructions: Move this check https://github.com/rust-lang/rust/blob/3b263ceb5cb89b6d53b5a03b47ec447c3a7f7765/src/librustdoc/config.rs#L372-L373
earlier to before `--passes=list` is processed: https://github.com/rust-lang/rust/blob/3b263ceb5cb89b6d53b5a03b47ec447c3a7f7765/src/librustdoc/config.rs#L330-L360
